darwin/future/core/client.py

```diff
@@ -235,6 +240,8 @@ def raise_for_darwin_exception(response: requests.Response) -> None:
     """
     if response.status_code == 200:
         return
+    if response.status_code == 400:
+        raise BadRequest(response)
```

Comment:
As covered in my comment in [PY-386 here](https://linear.app/v7labs/issue/PY-386/core-edit-items-move-to-folder#comment-4929616a), we probably want to catch the 400 errors this endpoint can throw in Core

To do this, I introduced the `BadRequest` class. This broke some existing tests that returned 400s, as previously they were expecting `HTTPError`, so updated each breaking test to expect BadRequest instead